sdk/internal/azurecli/azcli.go

```diff
@@ -59,68 +48,107 @@ func CheckAzVersion(minVersion string, nextMajorVersion *string) error {
 	return nil
 }
 
-// GetDefaultSubscriptionID tries to determine the default subscription
-func GetDefaultSubscriptionID() (string, error) {
-	var account struct {
-		SubscriptionID string `json:"id"`
-	}
-	err := JSONUnmarshalAzCmd(&account, "account", "show")
-	if err != nil {
-		return "", fmt.Errorf("obtaining subscription ID: %s", err)
-	}
-
-	return account.SubscriptionID, nil
-}
-
 // CheckTenantID validates the supplied tenant ID, and tries to determine the default tenant if a valid one is not supplied.
-func CheckTenantID(tenantId string) (string, error) {
+func CheckTenantID(tenantId string) (*string, error) {
 	validTenantId, err := regexp.MatchString("^[a-zA-Z0-9._-]+$", tenantId)
 	if err != nil {
-		return "", fmt.Errorf("could not parse tenant ID %q: %s", tenantId, err)
+		return nil, fmt.Errorf("could not parse tenant ID %q: %s", tenantId, err)
 	}
 
 	if !validTenantId {
```

Comment:
_nit / not a blocker_ but since `GetDefaultSubscriptionID` exists perhaps this method wants splitting into `GetDefaultTenantID` and `CheckTenantID`?

Reply:
done :)